examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/OfferingFragment.kt

```diff
@@ -138,4 +192,110 @@ class OfferingFragment : Fragment(), PackageCardAdapter.PackageCardAdapterListen
+                    requireActivity(),
+                    currentPackage.product,
+                    purchaseOption,
+                    purchaseErrorCallback,
+                    successfulPurchaseCallback
+                )
+            }
+        } else {
+            if (purchaseOption == null) {
+                Purchases.sharedInstance.purchasePackageWith(
+                    requireActivity(),
+                    currentPackage,
+                    upgradeInfo,
+                    purchaseErrorCallback,
+                    successfulUpgradeCallback
+                )
+            } else {
+                Purchases.sharedInstance.purchaseSubscriptionOptionWith(
+                    requireActivity(),
+                    currentPackage.product,
+                    purchaseOption,
+                    upgradeInfo,
+                    purchaseErrorCallback,
+                    successfulUpgradeCallback
+                )
+            }
+        }
+    }
+
+    private fun toggleLoadingIndicator(isLoading: Boolean) {
+        binding.purchaseProgress.visibility = if (isLoading) View.VISIBLE else View.GONE
+    }
+
+    private fun showOldSubIdPicker(callback: (String?) -> Unit) {
+        val activeSubIds = activeSubscriptions.map { it.split(":").first() } // TODOBC5 remove sub Id parsing
+        if (activeSubIds.isEmpty()) {
+            Toast.makeText(
+                requireContext(),
+                "Cannot upgrade without an existing active subscription.",
+                Toast.LENGTH_LONG
+            ).show()
+            toggleLoadingIndicator(false)
+            callback(null)
+            return
+        }
+
+        var selectedUpgradeSubId: String? = null
+        MaterialAlertDialogBuilder(requireContext())
+            .setTitle("Choose which active sub to switch from")
+            .setSingleChoiceItems(activeSubIds.toTypedArray(), 0) { _, which ->
+                selectedUpgradeSubId = activeSubIds[which]
+            }
+            .setPositiveButton("Continue") { dialog, _ ->
+                dialog.dismiss()
+                callback(selectedUpgradeSubId ?: activeSubIds[0])
+            }
+            .setNegativeButton("Cancel purchase") { dialog, _ ->
+                dialog.dismiss()
+                toggleLoadingIndicator(false)
+                callback(null)
+            }
+            .show()
+    }
+
+    private fun showProrationModePicker(callback: (Int?) -> Unit) {
+        val prorationModeOptions = mapOf(
+            0 to "None",
+            ProrationMode.IMMEDIATE_WITH_TIME_PRORATION to "IMMEDIATE_WITH_TIME_PRORATION",
+            ProrationMode.IMMEDIATE_AND_CHARGE_PRORATED_PRICE to "IMMEDIATE_AND_CHARGE_PRORATED_PRICE",
+            ProrationMode.IMMEDIATE_WITHOUT_PRORATION to "IMMEDIATE_WITHOUT_PRORATION",
+            ProrationMode.DEFERRED to "DEFERRED",
+            ProrationMode.IMMEDIATE_AND_CHARGE_FULL_PRICE to "IMMEDIATE_AND_CHARGE_FULL_PRICE"
+        )
```

Comment:
To avoid magic strings you can do
```suggestion
        val prorationModeOptions = mapOf(
            0 to "None",
            ProrationMode.IMMEDIATE_WITH_TIME_PRORATION to ProrationMode::IMMEDIATE_WITH_TIME_PRORATION.name,
            ProrationMode.IMMEDIATE_AND_CHARGE_PRORATED_PRICE to ProrationMode::IMMEDIATE_AND_CHARGE_PRORATED_PRICE.name,
            ProrationMode.IMMEDIATE_WITHOUT_PRORATION to ProrationMode::IMMEDIATE_WITHOUT_PRORATION.name,
            ProrationMode.DEFERRED to ProrationMode::DEFERRED.name,
            ProrationMode.IMMEDIATE_AND_CHARGE_FULL_PRICE to ProrationMode::IMMEDIATE_AND_CHARGE_FULL_PRICE.name
        )
```